pkg/cmd/psa/psa.go

```diff
@@ -0,0 +1,243 @@
+package psa
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"os"
+
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/klog"
+	psapi "k8s.io/pod-security-admission/api"
+
+	"github.com/spf13/cobra"
+)
+
+// PSAOptions contains all the options and configsi for running the PSA command.
+type PSAOptions struct {
+	kubeconfig string
+	level      string
+
+	client   *kubernetes.Clientset
+	warnings *warningsHandler
+}
+
+var (
+	psaExample = `
+	# pick a location of the kubeconfig file that is not ~/.kube/config.
+	%[1]s psa-check --kubeconfig /home/user/Documents/clusters/kubeconfig
+
+	# pick a pod-security.kubernetes.io/enforce: restricted.
+	%[1]s psa-check --level restricted
+`
+
+	anyLevel    = struct{}{}
+	validLevels = map[string]struct{}{
+		string(psapi.LevelPrivileged): anyLevel,
+		string(psapi.LevelBaseline):   anyLevel,
+		string(psapi.LevelRestricted): anyLevel,
+	}
+)
+
+// NewCmdPSA creates a cobra.Command that is hooked up to the kubectl-dev_tool.
+func NewCmdPSA(parentName string, streams genericclioptions.IOStreams) *cobra.Command {
+	o := PSAOptions{}
+
+	cmd := cobra.Command{
+		Use:     "psa-check",
+		Short:   "Check namespaces for a higher pod security enforce value.",
+		Example: fmt.Sprintf(psaExample, parentName),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			if err := o.Validate(); err != nil {
+				return err
+			}
+			if err := o.Complete(); err != nil {
```

Comment:
Generally `Complete` takes place before `Validate`. Because `Complete` is used to map flags to `PSAOptions` object and `Validate` validates `PSAOptions`.

Reply:
Yes, a blunder 😅